src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableValidator.java

```diff
@@ -470,6 +469,8 @@ private void validateTableCharacterContent(FieldValueValidator fieldValueValidat
     ArrayList<Integer> lineLengthsArray = new ArrayList<>(0);
     ArrayList<Long> lineNumbersArray = new ArrayList<>(0);
 
+    line = tableIsFixedLength ? this.currentTableReader.readNextFixedLine() : this.currentTableReader.readNextLine();
```

Comment:
It feels like the class of `currentTableReader` should provide a single API `readNextLine` and it itself figures out if it needs to provide a fixed line or a next line rather than relying on clients of the object to do it.

But I know, I know, I sound like a broken record. This is just the way "validate" is architected! 😝

Reply:
Actually, no. It is not the reader that thinks it is fixed length. It is the XML definition that thinks it should be then the test code tells the reader to do a fixed length read rather delimited based on 3 values in the XML - touch more messed up than you were giving it credit for.

Your points are correct: Yes, some XML adapter should be asked if it is fixed length. Or, yes, the reader should be more complete and use the XML do the necessary read (jparser). Yes, nothing will change because "do not fix it if it is not broken".

Reply:
@nutjob4life @al-niessner if you have a proposal for an architectural improvement to the software, feel free to create a ticket for it, and we can see if it becomes a higher priority in the future. I am all good with "do not fix it if it is not broken", until the technical debt becomes too high for us to manage. @al-niessner let me know when you think we have hit that tipping point.